static/src/javascripts/projects/common/modules/experiments/ab-tests.js

```diff
@@ -1,36 +1,132 @@
+import {
+    allRunnableTests,
+    firstRunnableTest,
+} from 'common/modules/experiments/ab-core';
+import {
+    registerCompleteEvents,
+    registerImpressionEvents,
+    trackABTests,
+} from 'common/modules/experiments/ab-ophan';
+import {
+    getParticipationsFromLocalStorage,
+    setParticipationsInLocalStorage,
+} from './ab-local-storage';
+import { getForcedParticipationsFromUrl } from './ab-url';
+import {
+    runnableTestsToParticipations,
+    testExclusionsWhoseSwitchExists,
+} from './ab-utils';
 
-export const TESTS: $ReadOnlyArray<ABTest> = [
-    getAcquisitionTest(),
+export const concurrentTests: $ReadOnlyArray<ABTest> = [
     commercialPrebidSafeframe,
     commercialAdVerification,
     commercialCmpCustomise,
     commercialAdMobileWebIncrease,
     commercialOutbrainNewids,
-].filter(Boolean);
-
-export const getActiveTests = (): $ReadOnlyArray<ABTest> =>
-    TESTS.filter(test => {
-        if (isExpired(test.expiry)) {
-            removeParticipation(test);
-            return false;
-        }
-        return true;
-    });
+];
+
+export const epicTests: $ReadOnlyArray<EpicABTest> = [
+    acquisitionsEpicUsTopTicker,
+    acquisitionsEpicAuPostOneMillion,
+    acquisitionsEpicFromGoogleDocOneVariant,
+    acquisitionsEpicFromGoogleDocTwoVariants,
+    acquisitionsEpicFromGoogleDocThreeVariants,
+    acquisitionsEpicFromGoogleDocFourVariants,
+    acquisitionsEpicFromGoogleDocFiveVariants,
+    acquisitionsEpicAusEnvCampaign,
+    acquisitionsEpicUSGunCampaign,
+    askFourEarning,
+    acquisitionsEpicAlwaysAskIfTagged,
+    acquisitionsEpicLiveblog,
+    acquisitionsEpicThankYou,
+];
+
+export const engagementBannerTests: $ReadOnlyArray<AcquisitionsABTest> = [
+    AcquisitionsBannerUsEoy,
+    AcquisitionsBannerAustraliaPostOneMillionTest,
+    AcquisitionsBannerGoogleDocTestOneVariant,
+    AcquisitionsBannerGoogleDocTestTwoVariants,
+    AcquisitionsBannerGoogleDocTestThreeVariants,
+    AcquisitionsBannerGoogleDocTestFourVariants,
+    AcquisitionsBannerGoogleDocTestFiveVariants,
+];
+
+// These are the tests which will actually take effect on this pageview.
+// Note that this is a subset of the potentially runnable tests,
+// because we only run one epic test and one banner test per pageview.
+// We memoize this because it can't change for a given pageview, and because getParticipations()
+// and isInVariant() depend on it and these are called in many places.
+export const getTestsToRun = memoize(
+    (): $ReadOnlyArray<Runnable<ABTest>> => {
+        const epicTest = firstRunnableTest(epicTests);
+        const engagementBannerTest = firstRunnableTest(engagementBannerTests);
+
+        return [
+            ...allRunnableTests(concurrentTests),
+            ...(epicTest ? [epicTest] : []),
+            ...(engagementBannerTest ? [engagementBannerTest] : []),
+        ];
+    }
+);
+
+// The tests which will take effect on this pageview,
+export const getParticipations = (): Participations =>
```

Comment:
`Participations` is a simplified data structure which is the one persisted in localStorage